go/vt/vttablet/tabletmanager/vreplication/replicator_plan.go

```diff
@@ -646,6 +654,24 @@ func appendFromRow(pq *sqlparser.ParsedQuery, buf *bytes2.Buffer, fields []*quer
 				buf.WriteString(sqltypes.NullStr)
 			} else {
 				vv := sqltypes.MakeTrusted(typ, row.Values[col.offset:col.offset+col.length])
+
+				if conversion, ok := tp.ConvertCharset[col.field.Name]; ok && col.length >= 0 {
```

Comment:
We don't want `col.length > 0` here? If there are no chars/bytes then I wouldn't think we need to do anything in this regard.

Reply:
Due to my bad English, I'm not sure if you mean we should use `col.length >= 0` or if you mean we shouldn't use `col.length >= 0`.

Just in case you mean the former, we do have `col.length >= 0` at the end of this line, in case you've missed it.
If you meant the latter, then `col.length >= 0`  in this context is an indicator that the value is not NULL, and we should test this or otherwise the conversion will break.

Reply:
@dbussink pointed out that you meant to highlight `> 0` rather than `>= 0`. Agreed, and fixed!